src/securityPolicy.c

```diff
@@ -363,13 +366,17 @@ security_policy_t policyForSignTxWitness(
 	DENY_UNLESS(is_valid_witness(pathSpec));
 
 	if (isSigningPoolRegistrationAsOwner) {
+
 		DENY_UNLESS(is_valid_stake_pool_owner_path(pathSpec));
+
+		WARN_UNLESS(bip44_hasReasonableAccount(pathSpec));
+
 	} else {
-		// TODO Perhaps we can relax this?
+		// witness for input or withdrawal
+
 		WARN_UNLESS(has_reasonable_account_and_address(pathSpec));
 
-		// TODO deny this? or check for depth in is_valid_witness?
-		WARN_IF(is_too_deep(pathSpec));
+		DENY_IF(is_too_deep(pathSpec));
```

Comment:
If this comes after `WARN_UNLESS`, wouldn't that mean that it deep paths with "unreasonable" account and address index would pass with a warning instead of being denied?